device/CMakeLists.txt

```diff
@@ -21,6 +21,7 @@ set(UMD_DEVICE_SRCS
     wormhole/wormhole_coordinate_manager.cpp
     pcie/pci_device.cpp
     hugepage.cpp
+    ../common/semver.cpp
```

Comment:
Should we instead compile a common lib, and then make device relying on common lib? @blozano-tt

Reply:
I can do that in this PR, but my thought is that we have opportunity for a broader cleanup.. today we have code in:

common/
src/ 
device/ <- most stuff lives here

I'm not sure what the best way to structure things is.  What do you think?  I am happy to create a ticket and defer this, equally happy to add a libumdcommon (or some such) for now.. let me know.

Reply:
... or we could keep common as header only, and move those small implementations that you have in .cpp file to a .h file as inline functions. Not sure what is the best way to keep things organised. Would like @blozano-tt 's insight here

Reply:
My rationale for putting it in .cpp was to keep the fmt include out of the header.

Reply:
I put it all into the header.  Please take another look.